### PR TITLE
rm scenario mapping for loan book prep

### PR DESCRIPTION
**Pull request by jacobvjk**
**This PR:**
- removes p4b to p4i scenario mapping (handled in data preparation) https://github.com/2DegreesInvesting/r2dii.stress.test.data/pull/91
- reads project agnostic files based on new flexibilisation: https://github.com/2DegreesInvesting/r2dii.stress.test.data/pull/91
- changes default in loan book preparation to 2021 production data, in order to cover the required forward looking timeframe for start year 2021 in the analysis
- removes hard coded scenario filtering in loan book preparation
- enables WEO2020 and GECO2019 scenarios
- sets default scenarios to WEO2020 SPS/SDS

**NOTE:**
This is dependent on https://github.com/2DegreesInvesting/r2dii.stress.test.data/pull/91 (replaced by smaller PRs all of which are merged)

### CODE REVIEW

File: data-raw/stress_test_arguments.R [hunk 1]
```diff
@@ -2,8 +2,8 @@
 stress_test_arguments <- tibble::tribble(
                       ~name,       ~type, ~default,               ~allowed,       ~min,       ~max,
                "asset_type", "character",       NA, "equity, bonds, loans",         NA,         NA,
-        "baseline_scenario", "character", "WEO2019_NPS", "WEO2019_NPS, ETP2017_NPS",         NA,         NA,
-           "shock_scenario", "character", "WEO2019_SDS", "WEO2019_SDS, ETP2017_SDS",         NA,         NA,
+        "baseline_scenario", "character", "WEO2020_SPS", "WEO2019_NPS, WEO2019_SPS, WEO2020_SPS, ETP2017_NPS, GECO2019_ref",         NA,         NA,
```
Comment (MirijaHa) [inline data-raw/stress_test_arguments.R:5]: FYI @jakubcervenka

File: data-raw/scenario_geography_x_ald_sector.R [hunk 2]
```diff
@@ -47,6 +47,51 @@ scenario_geography_x_ald_sector <- tibble::tribble(
   "Russia",      "Power",     "WEO2019_NPS",
```
Comment (MirijaHa) [inline data-raw/scenario_geography_x_ald_sector.R:47]: I would suggest sorting this by scen_geo for easier understanding. You could easily generated respective tibble with datapasta

File: R/format_loanbook_st.R [hunk 3]
```diff
@@ -123,7 +123,7 @@ format_loanbook_st <- function(data,
     dplyr::select(-.data$scen_tech_prod)
 
   scen <- results_loanbook %>%
-    dplyr::filter(.data$scenario %in% p4b_scenarios_lookup) %>%
+    dplyr::filter(stringr::str_detect(.data$scenario, "target_")) %>%
```
Comment (MirijaHa) [inline R/format_loanbook_st.R:126]: comment pls

File: data-raw/stress_test_arguments.R [hunk 4]
```diff
@@ -2,8 +2,8 @@
 stress_test_arguments <- tibble::tribble(
                       ~name,       ~type, ~default,               ~allowed,       ~min,       ~max,
```
Comment (MirijaHa) [inline data-raw/stress_test_arguments.R:3]: please run devtools::test or check and update the snapshot test.

**Review by MirijaHa — CHANGES_REQUESTED** [review]
failing devtools checks, compare comment

File: R/format_loanbook_st.R (hunk 3 above)
Comment (jacobvjk) [reply R/format_loanbook_st.R:128]: comment added

File: data-raw/stress_test_arguments.R (hunk 4 above)
Comment (jacobvjk) [reply data-raw/stress_test_arguments.R:3]: updated the one for the arguments, some of the other fails may not be actual errors. Will confirm on Monday

File: data-raw/scenario_geography_x_ald_sector.R (hunk 2 above)
Comment (jacobvjk) [reply data-raw/scenario_geography_x_ald_sector.R]: done

File: R/lookup.R [hunk 5]
```diff
@@ -40,6 +40,7 @@ equity_market_filter_lookup <- "GlobalMarket"
 p4i_p4b_sector_technology_lookup <- tibble::tribble(
   ~sector_p4b,   ~technology_p4b,             ~sector_p4i,    ~technology_p4i,
   "automotive",  "electric",                 "Automotive",   "Electric",
+  "automotive",  "fuelcell",                 "Automotive",   "FuelCell",
```
Comment (MirijaHa) [inline R/lookup.R:43]: 1. change of this mapper makes calculation of ETP2017 fail
2. This introduces an inconsistency between this mapper and its mirror in st data prep repo. this is a bit of a problem. In data prep we use this mapper to infer if sectors are complete

I synch we should quickly synch on how to resolve this.

**Review by MirijaHa — CHANGES_REQUESTED** [review]
Failing for etp2017, compare comment

**Review by MirijaHa — APPROVED** [review]
🔥

**Pull request merged**